src/doctors/doctors.resolver.ts

```diff
@@ -35,11 +36,13 @@ export class DoctorsResolver {
 
   @ResolveField('user')
   async getUser(@Parent() doctor: Doctor) {
-    return await this.usersService.findOne(doctor.user.id);
+    return await this.usersService.findOneById(doctor.user.id);
   }
 
   @ResolveField('specialization')
   async getSpecialization(@Parent() doctor: Doctor) {
-    return await this.specializationsService.findOne(doctor.specialization.id);
+    return await this.specializationsService.findOneById(
+      doctor.specialization.id,
+    );
   }
```

Comment:
It's strange that to query specialization it should already be in doctor. In that case, you can just return it. Or you can use a doctor.specializationId field.